controllers/vspherevm_controller.go

```diff
@@ -269,6 +271,25 @@ func (r vmReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.R
 		if err := vmContext.Patch(ctx); err != nil {
 			reterr = kerrors.NewAggregate([]error{reterr, err})
 		}
+
+		// Wait until VSphereVM is updated in the cache if the `.Status.TaskRef` field changes.
+		// Note: We have to do this because otherwise using a cached client in current state could
+		// return a stale state of a VSphereVM we just patched (because the cache might be stale).
```

Comment:
Let's add one more line about that this can lead to duplicate tasks being triggered (e.g. for VM deletion)